Gauntlet.toml

```diff
@@ -46,658 +44,309 @@ commit_hash = "ee08634f28810e5d6fd1a904fc83f4e67821550e"
+message = '/samtools.wdl:326:44: error: unknown escape sequence `\.`'
 
-[[concerns]]
-kind = "ValidationFailure"
+[[diagnostics]]
 document = "biowdl/tasks:/samtools.wdl"
-message = '''[v1::E001] invalid escape character: '\.' (414:44-414:46)'''
+message = '/samtools.wdl:414:44: error: unknown escape sequence `\.`'
 
-[[concerns]]
-kind = "ValidationFailure"
+[[diagnostics]]
 document = "biowdl/tasks:/samtools.wdl"
-message = '''[v1::E001] invalid escape character: '\.' (458:49-458:51)'''
+message = '/samtools.wdl:458:49: error: unknown escape sequence `\.`'
 
-[[concerns]]
-kind = "ValidationFailure"
+[[diagnostics]]
 document = "biowdl/tasks:/samtools.wdl"
-message = '''[v1::E001] invalid escape character: '\.' (470:60-470:62)'''
+message = '/samtools.wdl:470:60: error: unknown escape sequence `\.`'
 
-[[concerns]]
-kind = "ValidationFailure"
+[[diagnostics]]
 document = "biowdl/tasks:/samtools.wdl"
-message = '''[v1::E001] invalid escape character: '\.' (80:42-80:44)'''
+message = '/samtools.wdl:80:42: error: unknown escape sequence `\.`'
 
-[[concerns]]
-kind = "ValidationFailure"
+[[diagnostics]]
 document = "biowdl/tasks:/umi-tools.wdl"
-message = '''[v1::E001] invalid escape character: '\.' (95:49-95:51)'''
+message = '/umi-tools.wdl:95:49: error: unknown escape sequence `\.`'
 
-[[concerns]]
-kind = "ValidationFailure"
+[[diagnostics]]
 document = "biowdl/tasks:/umi.wdl"
-message = '''[v1::E001] invalid escape character: '\.' (39:60-39:62)'''
+message = '/umi.wdl:39:60: error: unknown escape sequence `\.`'
+
+[[diagnostics]]
+document = "broadinstitute/palantir-workflows:/HaplotypeMap/BuildHaplotypeMap.wdl"
+message = "/HaplotypeMap/BuildHaplotypeMap.wdl:17:1: error: a WDL document must start with a version statement"
 
-[[concerns]]
-kind = "ValidationFailure"
+[[diagnostics]]
 document = "broadinstitute/warp:/pipelines/skylab/scATAC/scATAC.wdl"
-message = "[v1::E003] duplicate runtime keys: 'cpu' (199:9-199:12, 203:9-203:12)"
+message = "/pipelines/skylab/scATAC/scATAC.wdl:203:9: error: duplicate key `cpu` in runtime section"
+
+[[diagnostics]]
+document = "broadinstitute/warp:/tasks/broad/GermlineVariantDiscovery.wdl"
+message = "/tasks/broad/GermlineVariantDiscovery.wdl:137:32: error: expected string, but found integer"
+
+[[diagnostics]]
+document = "broadinstitute/warp:/tasks/broad/GermlineVariantDiscovery.wdl"
+message = "/tasks/broad/GermlineVariantDiscovery.wdl:65:32: error: expected string, but found integer"
+
+[[diagnostics]]
+document = "broadinstitute/warp:/tasks/broad/UltimaGenomicsWholeGenomeGermlineTasks.wdl"
+message = "/tasks/broad/UltimaGenomicsWholeGenomeGermlineTasks.wdl:814:27: error: expected string, but found integer"
+
+[[diagnostics]]
+document = "broadinstitute/warp:/tasks/broad/UltimaGenomicsWholeGenomeGermlineTasks.wdl"
+message = "/tasks/broad/UltimaGenomicsWholeGenomeGermlineTasks.wdl:866:27: error: expected string, but found integer"
```

Comment:
I think this is a bug on our part. Or I'm confused. The below code looks like valid WDL to me.

https://github.com/broadinstitute/warp/blob/develop/tasks/broad/UltimaGenomicsWholeGenomeGermlineTasks.wdl#L866

Reply:
According to the spec:

> The result of evaluating an expression in a placeholder must ultimately be converted to a string in order to take the place of the placeholder in the command script

...

> Expression placeholder options are `option="value"` pairs that precede the expression within an expression placeholder and customize the interpolation of the WDL value into the containing string expression.

Making it seem that the option's value must be a quoted string as the result of the interpolation will always be a string anyway.

This makes sense for `sep` as it's effectively the separator argument to the `sep` function; it also makes sense for `true` and `false` as they're just replacement strings for those values respectively.

It goes on to say:

> If an expression within a placeholder evaluates to None, then the placeholder is replaced by the empty string.

which makes sense as well (i.e. the default for the `default` placeholder option is `""`).

However, it adds this about the `default` option specifically:

> **The type of the default value must match the type of the expression**

The last one makes absolutely no sense; the default value shouldn't need match the type of the expression; it's simply what is used _instead of the empty string_ when the expression evaluates to `None`. 

Despite the contradictory elements of the spec, I think accepting anything but a string for the `default` option's value should not be legal, otherwise a parser would have to accept, at the minimum, a subset of _literal expression_ at that position, which seems incorrect and definitely not spelled out in the spec like they do for metadata values.

Given that placeholder options are deprecated and there were only a few occurrences of not using a quoted value in the gauntlet run, I was thinking the current behavior and a "c’est la vie" approach would be best here.

Reply:
In fact, I think we should be linting the use of placeholder options as part of the "deprecation" rule.

Reply:
FWIW, an official parser does accept a [string or number](https://github.com/openwdl/wdl-parsers/blob/main/antlr4/1.1/WdlV1_1Parser.g4#L50) for the `default` placeholder option; I wonder if that's simply because it was observed in the wild rather than a result of interpreting the spec. If those are the only two accepted literal expressions at that position, the spec should clearly say so.

A literal interpretation of the spec (specifically the part about `the type of the default value must match the type of the expression`) would require `None` to be accepted here too because if the default value's type must match and the type of placeholder expression is required to be *optional*, then the default value's type must also be *optional*, allowing for `None` as a value. Obviously that would be a useless `default` option value.

At any rate, I'm also happy to fix the parser to accept numbers for the default option, despite it being converted to string _anyway_ as the result of the interpolation.

Reply:
Opened https://github.com/stjude-rust-labs/wdl/issues/77 to track this.